modules/diskq/qdisk.c

```diff
@@ -569,6 +569,23 @@ qdisk_remove_head(QDisk *self)
   return TRUE;
 }
 
+gboolean
+qdisk_serialize_msg(QDisk *self, LogMessage *msg, GString *serialized)
+{
+  gboolean result = TRUE;
+  SerializeArchive *sa = serialize_string_archive_new(serialized);
+
+  if (!log_msg_serialize(msg, sa, self->options->compaction ? LMSF_COMPACTION : 0))
+    {
+      g_string_truncate(serialized, 0);
```

Comment:
[optional] truncating the serialized method not really needed, by default I would expect that if a function returns with an error, it leaves the `serialized` in an unspecified (but valid) state[1]. So I need to for example free it, but cannot assume much about the content.
As an alternative to the unspecified, that it does not change the `serialized` string in case of failure. This is good, when you want to process multiple messages and batch into one string.

That's the reasoning why I would not do truncate here.

[1] One could say that always truncating is one of the unspecified and valid state, which is true. But not having code to reach that always nice.

Reply:
I removed the truncate from here, and removed the `log_msg_unref()` from the deserialization, too.